packages/api/src/external/commonwell/document/document-query.ts

```diff
@@ -747,11 +752,20 @@ async function triggerDownloadDocument({
     location: fileInfo.fileLocation,
   };
 
-  const result = await docDownloader.download({ document, fileInfo: adjustedFileInfo, cxId });
-  return {
-    ...result,
-    isNew: true,
-  };
+  try {
+    const result = await docDownloader.download({ document, fileInfo: adjustedFileInfo, cxId });
+    return {
+      ...result,
+      isNew: true,
+    };
+    //eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (error: any) {
+    if (error.response?.status === httpStatus.NOT_FOUND) {
```

Comment:
Did you test the `DocumentDownloaderLambda` will return 404 when the lambda gets 404 from CW and throws it?

By the look of this PR's code this code here will never execute. See the message I sent you on Slack (edited here for clarity)

> even if the lambda (errors) while executing, we get a 200 from the request, since it actually worked to call the lambda
> check `getLambdaResultPayload` and its usage to see that in action, that function ~~is~~ (was built) so we get errors from (calling) lambdas

Reply:
Tested now. It throws an error